### PR TITLE
Refactor code in MapModal component

### PR DESCRIPTION
**Pull request by Marina20V**
## Code reviewers

- [ ] @Depesi 

### Second Level Review

- [ ] @VadymKharyton 

## Summary of issue
An old component need to be refactored

## Summary of change
Refactor code  in the component

## Testing approach
Testing in browser

### CODE REVIEW

File: src/components/event/MapModal/Map-modal.js [hunk 1]
```diff
@@ -1,58 +1,54 @@
...
+        const startValue = initialize({
             radius: 8,
             selectedPos: { latitude: null, longitude: null }
         })
-        if (this.props.values.selectedPos.latitude != null)
-            return startValue;
-        else
-            return startValue && this.setState({ open: false })  
+
+        (values.selectedPos.latitude != null) ? startValue :
```
[the remark this answers is not in the document]
Comment (Stanislav-Vyhnan) [reply src/components/event/MapModal/Map-modal.js:29]: Change !=/== on !==/=== everywhere on file

[the remark this answers is not in the document]
Comment (Marina20V) [reply src/components/event/MapModal/Map-modal.js:29]: > I think, 'return' is need to be here

The ternary operator evaluates an expression and expressions can't contain a return.

[the remark this answers is not in the document]
Comment (Marina20V) [reply src/components/event/MapModal/Map-modal.js]: > Change !=/== on !==/=== everywhere on file

fixed it.

File: src/components/event/MapModal/Map-modal.js [hunk 2]
```diff
@@ -61,44 +57,43 @@ class MapModal extends Component {
...
                         <div>
                             {
-                                this.props.values &&
-                                this.props.values.selectedPos != undefined &&
-                                this.props.values.selectedPos.latitude != undefined &&
-                                this.props.values.selectedPos.longitude != undefined &&
+                                // values &&
+                                // values.selectedPos != undefined &&
+                                values?.selectedPos?.latitude !== undefined &&
+                                values?.selectedPos?.longitude !== undefined &&
```
Comment (VadymKharyton) [inline src/components/event/MapModal/Map-modal.js:63]: Please verify if we can simplify these conditions to not explicitly check for `null` or `undefined` values

Comment (VadymKharyton) [inline src/components/event/MapModal/Map-modal.js:63]: Please verify if we can simplify these conditions to not explicitly check for null or undefined values

File: src/components/event/MapModal/Map-modal.js [hunk 3]
```diff
@@ -1,58 +1,54 @@
...
+                <Button variant="outlined" fullWidth={true} color="primary" onClick={handleClickOpen}>
+                    {constants.FILTER_BY_LOCATION}
                 </Button>
-                <Dialog fullWidth={true} open={this.state.open} onClose={this.handleClose} aria-labelledby="form-dialog-title">
-                    <DialogTitle id="form-dialog-title">Filter by location</DialogTitle>
+                <Dialog fullWidth={true} open={open} onClose={handleClose} aria-labelledby="form-dialog-title">
+                    <DialogTitle id="form-dialog-title">{constants.FILTER_BY_LOCATION}</DialogTitle>
                     <DialogContent>
-                        {this.props.values && this.props.values.radius &&
+                        {values && values.radius &&
```
Comment (VadymKharyton) [inline src/components/event/MapModal/Map-modal.js:44]: Use shortcut `values?.radius`

File: src/components/event/MapModal/Map-modal.js [hunk 4]
```diff
@@ -1,58 +1,54 @@
...
+        const startValue = initialize({
             radius: 8,
             selectedPos: { latitude: null, longitude: null }
         })
-        if (this.props.values.selectedPos.latitude != null)
-            return startValue;
-        else
-            return startValue && this.setState({ open: false })  
+
+        (values.selectedPos.latitude !== null) ? startValue :
```
Comment (VadymKharyton) [inline src/components/event/MapModal/Map-modal.js:29]: Please verify if we can simplify these conditions to not explicitly check for null or undefined values

File: src/components/event/MapModal/Map-modal.js [hunk 5]
```diff
@@ -1,58 +1,54 @@
...
+import constants from '../../../constants/mapModal';
 
-class MapModal extends Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            open: false
-        }
-    }
+export const MapModal = (initialize,values) => {
```
Comment (VadymKharyton) [inline src/components/event/MapModal/Map-modal.js:14]: There is only one parameter in functional component(`props`). Use destruct operator in get these values: `{initialize,values}`

File: src/components/event/MapModal/Map-modal.js [hunk 6]
```diff
@@ -61,44 +57,40 @@ class MapModal extends Component {
                         }
                         <div>
                             {
-                                this.props.values &&
-                                this.props.values.selectedPos != undefined &&
-                                this.props.values.selectedPos.latitude != undefined &&
-                                this.props.values.selectedPos.longitude != undefined &&
+                                !!values.selectedPos.latitude &&
+                                !!values.selectedPos.longitude &&
```
Comment (VadymKharyton) [inline src/components/event/MapModal/Map-modal.js:61]: Same suggestion as above

File: src/components/event/MapModal/Map-modal.js [hunk 7]
```diff
@@ -61,44 +57,40 @@ class MapModal extends Component {
...
+                                    <p>{constants.LONGITUDE} {values.selectedPos.longitude}</p>
+                                    <DisplayMap location={{ ...values.selectedPos }}/>
                                 </div>
                             }
                             {
-                                this.props.values &&
-                                this.props.values.selectedPos.latitude == null &&
-                                this.props.values.selectedPos.longitude == null &&
+                                !!values.selectedPos.latitude  &&
+                                !!values.selectedPos.longitude &&
```
Comment (VadymKharyton) [inline src/components/event/MapModal/Map-modal.js:71]: Same suggestion as above

File: src/components/event/MapModal/Map-modal.js [hunk 8]
```diff
@@ -1,58 +1,54 @@
...
+        const startValue = initialize({
             radius: 8,
             selectedPos: { latitude: null, longitude: null }
         })
-        if (this.props.values.selectedPos.latitude != null)
-            return startValue;
-        else
-            return startValue && this.setState({ open: false })  
+
+        !!values.selectedPos.latitude ? startValue :
```
Comment (VadymKharyton) [inline src/components/event/MapModal/Map-modal.js:29]: There is no need to make extra conversion of this variable to the boolean types. We can just to check if variable have some value: `values.selectedPos.latitude ? ... : ...`

Comment (VadymKharyton) [reply src/components/event/MapModal/Map-modal.js:29]: Add please destruct `latitude` property somewhere in the component to not refer to this variable with full path(`values.selectedPos.latitude`) each time(I see, that it used a lot of times in this component).

File: src/components/event/MapModal/Map-modal.js [hunk 9]
```diff
@@ -1,58 +1,54 @@
-﻿import React, { Component } from 'react';
+﻿import React, { useState } from 'react';
 import Button from '@material-ui/core/Button';
 import Dialog from '@material-ui/core/Dialog';
 import DialogActions from '@material-ui/core/DialogActions';
 import DialogContent from '@material-ui/core/DialogContent';
 import DialogTitle from '@material-ui/core/DialogTitle';
-import { LocationMapWithCircle } from '../../helpers/form-helpers/location';
 import { Field } from 'redux-form';
+
```
Comment (andrii-kuzmichov) [inline src/components/event/MapModal/Map-modal.js:8]: remove empty row

File: src/constants/mapModal.js [hunk 10]
```diff
@@ -0,0 +1,14 @@
+const constants = {
```
Comment (andrii-kuzmichov) [inline src/constants/mapModal.js:1]: Change name of constant

File: src/components/event/MapModal/Map-modal.js [hunk 11]
```diff
@@ -61,44 +57,40 @@ class MapModal extends Component {
...
                             }
                             {
-                                this.props.values &&
-                                this.props.values.selectedPos.latitude == null &&
-                                this.props.values.selectedPos.longitude == null &&
+                                !!values.selectedPos.latitude  &&
+                                !!values.selectedPos.longitude &&
                                 <div>
-                                    <p>Choose position on the Map!</p>
+                                    <p>{constants.CHOOSE_MAP_POSITION}</p>
```
Comment (andrii-kuzmichov) [inline src/components/event/MapModal/Map-modal.js:73]: Use destructuring for constants in each case

**Pull request closed without merging**